src/NHibernate/Cfg/XmlHbmBinding/CollectionBinder.cs

```diff
@@ -192,6 +192,7 @@ private void BindPrimitiveArray(HbmPrimitiveArray arrayMapping, PrimitiveArray m
 				var oneToMany = new OneToMany(model.Owner);
 				model.Element = oneToMany;
 				BindOneToMany(oneToManyMapping, oneToMany);
+				oneToMany.PropertyName = collectionMapping.Name;
```

Comment:
This fix doesn't affect `one-to-many` case so I think we should undo this change